edi_oca/models/edi_exchange_record.py

```diff
@@ -209,7 +211,15 @@ def record(self):
             return None
         if not self.model and self.parent_id:
             return self.parent_id.record
-        return self.env[self.model].browse(self.res_id).exists()
+        rec = self.env[self.model].browse(self.res_id).exists()
+        if (
+            rec
+            and "company_id" in rec
+            and rec.sudo().company_id.id not in self.env.companies.ids
+        ):
+            # Return empty recordset if company does not match
+            rec = rec[:0]
```

Comment:
I wonder if we could miss another case: when the company_id id not set. Here latest proposal:
```suggestion
        rec = self.env[self.model].browse(self.res_id).exists()
        if rec and "company_id" in rec:
            rec_company = rec.sudo().company_id
            if rec_company and rec_company not in self.env.companies:
                # Return empty recordset if company does not match
                rec = rec[:0]
```